memory/vatest.py.data/va_test.c

```diff
@@ -324,10 +324,50 @@ void mixed_position_alterhuge(int chunks1, int chunks2, int chunks3)
 	mmap_chunks_higher(chunks2/2, MAP_HUGETLB);
 }
 
+int meminfo(char cmd[])
+{
+        char buffer[100];
+        int num = 0;
+        FILE* file = popen(cmd, "r");
+        fgets(buffer, 100, file);
+        char * token = strtok(buffer, " ");
+        while( token != NULL )
+        {
+                num = atoi(token);
+                if (num == 0 && token != '0');
+                else{
+                        //printf("Number is: %i", num);
+                        token = NULL;
+                        return num;
+                }
+
+                token = strtok(NULL, " ");
+        }
+}
+
+void mem_checkpoint()
+{
+        int swap_free_space, total_available_mem, total_free_space, total_free_mm, map_chunk_size_gb;
+        char swap_cmd[] = "cat /proc/meminfo | grep -i swapfree";
+        char mem_total_cmd[]= "cat /proc/meminfo | grep -i memavailable";
+        swap_free_space = meminfo(swap_cmd);
+        total_available_mem = meminfo(mem_total_cmd);
+        total_free_space = swap_free_space + total_available_mem;
+        total_free_mm = total_free_space/(1024*1024);
+        map_chunk_size_gb = MAP_CHUNK_SIZE / 1024 / 1024 / 1024;
+        if(map_chunk_size_gb > total_free_mm)
+        {
+                unsigned long mm_aloc = total_free_mm - 2;
+                unsigned long map_chunk_size = mm_aloc*1024*1024*1024;
```

Comment:
Do we really need these 2 additional variables. We should be able to change this to
MAP_CHUNK_SIZE = (total_free_mm - 2)*1024*1024*1024